backend/danswer/key_value_store/store.py

```diff
@@ -0,0 +1,79 @@
+import json
+from collections.abc import Iterator
+from contextlib import contextmanager
+from typing import cast
+
+from sqlalchemy.orm import Session
+
+from danswer.db.engine import get_session_factory
+from danswer.db.models import KVStore
+from danswer.key_value_store.interface import JSON_ro
+from danswer.key_value_store.interface import KeyValueStore
+from danswer.key_value_store.interface import KvKeyNotFoundError
+from danswer.redis.redis_pool import get_redis_client
+
+REDIS_KEY_PREFIX = "danswer_kv_store:"
+
+
+class PgRedisKVStore(KeyValueStore):
+    def __init__(self) -> None:
+        self.redis_client = get_redis_client()
+
+    @contextmanager
+    def get_session(self) -> Iterator[Session]:
+        factory = get_session_factory()
+        session: Session = factory()
+        try:
+            yield session
+        finally:
+            session.close()
+
+    def store(self, key: str, val: JSON_ro, encrypt: bool = False) -> None:
+        # Not encrypted in Redis, but encrypted in Postgres
+        self.redis_client.set(REDIS_KEY_PREFIX + key, json.dumps(val))
```

Comment:
we should handle a potential RedisError exception here.  Also note that without using a TTL the error handling gets significantly trickier as we run into potential consistency issues that won't resolve themselves if there is a spurious exception.